drivers/flash/flash_shell.c

```diff
@@ -52,12 +52,18 @@ static int cmd_erase(const struct shell *shell, size_t argc, char *argv[])
 	if (argc > 2) {
 		size = strtoul(argv[2], NULL, 16);
 	} else {
-#if defined(CONFIG_SOC_FLASH_NRF)
-		size = NRF_FICR->CODEPAGESIZE;
-#else
-		error(shell, "Missing size.");
-		return -EINVAL;
-#endif
+		struct flash_pages_info info;
+
+		result = flash_get_page_info_by_offs(flash_dev, page_addr,
```

Comment:
@aurel32 I'm wondering if the user does not specify the erase size and we use the page size retrieved via flash page layout that we should also change the page address to the start of the page returned by ```flash_get_page_info_by_offs```

Reply:
I am not sure about that, the problem is that the API is not clear about that and drivers have different behaviours. For example, the nRF and the Atmel SAM drivers expect to get the address aligned to the page size and the size to be a multiple of size. On the other hand the STM32 driver erases all pages that contain the address [address, address + size[.

Maybe leave that unchanged for now?

Reply:
I prefer to keep this as is. I was not aware that stm32 driver behaves like that (It's rather unwanted behavior).

Reply:
User should know what he/she is doing.

Reply:
ok I will keep it as-is, thx for the input